pkg/operation/botanist/dns.go

```diff
@@ -87,7 +97,16 @@ func (b *Botanist) DeployInternalDNS(ctx context.Context) error {
 // the internal DNSProvider and DNSEntry.
 func (b *Botanist) InternalDNS() component.Deployer {
 	if b.NeedsInternalDNS() {
+		if b.isRestorePhase() {
```

Comment:
The code in this if is exactly the same as the one after it. Unless the order of deployment is changed (see other comments), this if is redundant and should be removed.